src/omv/modules/py_image.c

```diff
@@ -768,14 +765,30 @@ static mp_obj_t py_image_unpack(uint n_args, const mp_obj_t *pos_args, mp_map_t
         }
     }
 
+    size_t shape[ULAB_MAX_DIMS] = {};
+    size_t strides[ULAB_MAX_DIMS] = {};
     int channels;
+    int ndim;
+
     switch (image->pixfmt) {
         case PIXFORMAT_GRAYSCALE: {
+            shape[ULAB_MAX_DIMS - 2] = image->h;
+            shape[ULAB_MAX_DIMS - 1] = image->w;
+            strides[ULAB_MAX_DIMS - 2] = image->w * dtype_size;
+            strides[ULAB_MAX_DIMS - 1] = dtype_size;
             channels = 1;
+            ndim = 2;
             break;
         }
         case PIXFORMAT_RGB565: {
+            shape[ULAB_MAX_DIMS - 3] = image->h;
+            shape[ULAB_MAX_DIMS - 2] = image->w;
+            shape[ULAB_MAX_DIMS - 1] = 3;
```

Comment:
Please use compound literals for all of these assignments:
```suggestion
            shape = (size_t []) {0, image->h, image->w, 3};
```

Reply:
Done